src/loaders/ccloudResourceLoader.ts

```diff
@@ -453,12 +457,33 @@ function restFlinkStatementToModelFlinkStatement(
   // but is really a SqlV1StatementSpec.
   const spec: SqlV1StatementSpec = restFlinkStatement.spec as SqlV1StatementSpec;
 
+  const status = restFlinkStatement.status;
+  const responseTraits = status.traits;
+
+  const modelTraits: FlinkStatementTraits | undefined = responseTraits
+    ? new FlinkStatementTraits({
+        sqlKind: responseTraits.sql_kind,
+        bounded: responseTraits.is_bounded,
+        appendOnly: responseTraits.is_append_only,
+        schema: responseTraits.schema,
+      })
+    : undefined;
+
   return new FlinkStatement({
     connectionId: CCLOUD_CONNECTION_ID,
     connectionType: ConnectionType.Ccloud,
     environmentId: restFlinkStatement.environment_id as EnvironmentId,
     computePoolId: spec.compute_pool_id!,
     name: restFlinkStatement.name,
-    status: restFlinkStatement.status.phase,
+    metadata: new FlinkStatementMetadata({
+      createdAt: restFlinkStatement.metadata.created_at,
+      updatedAt: restFlinkStatement.metadata.updated_at,
+    }),
+    status: {
+      phase: restFlinkStatement.status.phase,
+      detail: restFlinkStatement.status.detail,
+      traits: modelTraits,
+      scalingStatus: restFlinkStatement.status.scaling_status,
+    },
```

Comment:
This makes me wonder if we should just start extending the existing openapi-generator / typescript-fetch interfaces and only adding the new fields we need, rather than summarizing/extracting existing fields. 🤔